helper/schema/field_reader_config.go

```diff
@@ -105,34 +105,67 @@ func (r *ConfigFieldReader) readField(
 }
 
 func (r *ConfigFieldReader) readMap(k string) (FieldReadResult, error) {
-	mraw, ok := r.Config.Get(k)
+	// We want both the raw value and the interpolated. We use the interpolated
+	// to store actual values and we use the raw one to check for
+	// computed keys.
+	mraw, ok := r.Config.GetRaw(k)
```

Comment:
The changes to this whole function are the saddest. The ConfigFieldReader before simply never marked the map as computed, ever. But this isn't the case. Unfortunately we have no real way to check if its computed other than checking every field.